### PR TITLE
Increase password strength

### PR DESCRIPTION
**Pull request by brucebolt**
- Change the minimum password length in user schema from four characters to eight characters (matches legacy CKAN) and update relevant error message
- Add hint text to explain how to create a stronger password

https://trello.com/c/NzGpQkop/403-increase-password-security-for-ckan

### CODE REVIEW

File: ckanext/datagovuk/templates/user/new_user_form.html [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+
+<form id="user-register-form" class="form-horizontal" action="" method="post">
+  {{ form.errors(error_summary) }}
+  {{ form.input("name", id="field-username", label=_("Username"), placeholder=_("username"), value=data.name, error=errors.name, classes=["control-medium"]) }}
+  {{ form.input("fullname", id="field-fullname", label=_("Full Name"), placeholder=_("Joe Bloggs"), value=data.fullname, error=errors.fullname, classes=["control-medium"]) }}
+  {{ form.input("email", id="field-email", label=_("Email"), type="email", placeholder=_("joe@example.com"), value=data.email, error=errors.email, classes=["control-medium"]) }}
+  {{ form.input("password1", id="field-password", label=_("Password"), type="password", placeholder="••••••••", value=data.password1, error=errors.password1, classes=["control-medium"], attrs={'pattern': '.{8,}', 'title': 'Password must be a minimum of 8 characters'}) }}
+  {{ form.input("password2", id="field-confirm-password", label=_("Confirm"), type="password", placeholder="••••••••", value=data.password2, error=errors.password1, classes=["control-medium"], attrs={'pattern': '.{8,}', 'title': 'Password must be a minimum of 8 characters'}) }}
```
Comment (andrewgarner) [inline ckanext/datagovuk/templates/user/new_user_form.html]: `error=errors.password2`?

Comment (brucebolt) [reply ckanext/datagovuk/templates/user/new_user_form.html]: You're absolutely right.  This was copied straight out of the core CKAN code to override in our plugin, so it must be incorrect there too.

Comment (andrewgarner) [reply ckanext/datagovuk/templates/user/new_user_form.html]: Nice open-source opportunity for you there then!

File: ckanext/datagovuk/controllers/user.py [hunk 2]
```diff
@@ -8,9 +8,28 @@ def _new_form_to_db_schema(self):
...
         if not c.user:
             h.redirect_to(locale=locale, controller='user', action='login',
                           id=None)
         user_ref = c.userobj.get_reference_preferred_for_uri()
         h.redirect_to(locale=locale, controller='user', action='dashboard_datasets')
+
+    def _get_form_password(self):
+        password1 = request.params.getone('password1')
+        password2 = request.params.getone('password2')
+        log.debug(password1)
```
Comment (andrewgarner) [inline ckanext/datagovuk/controllers/user.py]: Hmm. We don't want to be logging passwords.

Comment (brucebolt) [reply ckanext/datagovuk/controllers/user.py]: Removed now.  Will have to make a new PR.

**Pull request merged**